fetch-object.c

```diff
@@ -4,14 +4,15 @@
 #include "strbuf.h"
 #include "transport.h"
 #include "fetch-object.h"
+#include "alloc.h"
 
 static void fetch_refs(const char *remote_name, struct ref *ref)
 {
 	struct remote *remote;
 	struct transport *transport;
 	int original_fetch_if_missing = fetch_if_missing;
 
-	fetch_if_missing = 0;
+	repo.fetch_if_missing = 0;
```

Comment:
You will need to adjust the signature of `fetch_refs()` to receive a pointer to the repository, the convention is to use `struct repository *r`.